backend/app/infrastructure/routes/rewards.py

```diff
@@ -217,10 +247,42 @@ def get(self, epoch: int):
 
 @ns.route("/estimated_budget")
 @ns.doc(
-    description="Returns estimated rewards budget available when GLM locked by given period of time"
+    description="Returns estimated rewards budget available when GLM locked by given number of full epochs"
 )
 class EstimatedUserBudget(OctantResource):
     @ns.expect(estimated_budget_request)
+    @ns.marshal_with(user_budget_with_matched_funding_model)
+    @ns.response(200, "Budget successfully retrieved")
+    def post(self):
+        no_epochs, glm_amount = ns.payload["numberOfEpochs"], int(
+            ns.payload["glmAmount"]
+        )
+        app.logger.debug(
+            f"Getting user estimated budget for {no_epochs} epochs and {glm_amount} GLM. Getting matched_funding based on previous epoch."
+        )
+
+        current_epoch_num = epochs.get_current_epoch()
```

Comment:
Please move this validations to the controller, where we already know the epoch number. Also, we shouldn't keep any logic here in routes, so please move `matched_funding` calculations to the service

Reply:
yeah, sorry - forgot to move it to the controllers